ibeam/src/gateway_client.py

```diff
@@ -68,6 +85,102 @@ def __init__(self,
+        Otherwise the self.secrets_source will be evaluated
+        to determine how to handle the value.
+
+        If self.secrets_source has been set to
+        SECRETS_SOURCE_ENV the os.environ value will be
+        returned as the secret value.
+
+        If self.secrets_source has been set to
+        SECRETS_SOURCE_FS then the os.environ value is
+        treated as a filesystem path.  The file will be read
+        as text and its contents returned as the secret
+        value.
+
+        Parameters:
+          name:
+            The identifier for the value, e.g.,
+            'IBEAM_ACCOUNT', 'IBEAM_PASSWORD', or
+            'IBEAM_KEY'.
+          lstrip:
+            If not None, strip these characters from the
+            left of the returned value (default: None).
+          rstrip:
+            If not None, strip these characters from the
+            right of the returned value (default: '\r\n')
+        Returns:
+          If the name is not defined in os.environ then None
+          is returned.
+
+          If self.secrets_source is SECRETS_SOURCE_ENV then
+          the os.environ value is returned as the secret
+          value.
+
+          If self.secrets_source is SECRETS_SOURCE_FS then
+          the os.environ value is treated as file path.  The
+          file is read as a text file and its contents
+          returned as the secret value.
+
+          If an error is encountered reading the file then
+          an error is logged and None is returned.
+        """
+        # read the environment value for name
+        value = os.environ.get(name)
+        if value is None:
+            # no key for this name, nothing to do
+            return None
+
+        if self.secrets_source == SECRETS_SOURCE_ENV:
+            # treat environment values as the secrets
+            # themselves
+            if lstrip is not None:
+                value = value.lstrip(lstrip)
+
+            if rstrip is not None:
+                value = value.rstrip(rstrip)
+
+            return value
+        elif self.secrets_source == SECRETS_SOURCE_FS:
+            # treat environment values as filesystem paths
+            # to the secrets
+            if not os.path.isfile(value):
+                _LOGGER.error(
+                    f'unable to read env value for {name}: value is not a file')
+                return None
+
+            try:
+                with open(value, mode='rt', encoding=self.encoding) as fh:
+                    secret = fh.read()
+
+                    if lstrip is not None:
+                        secret = secret.lstrip(lstrip)
+
+                    if rstrip is not None:
+                        secret = secret.rstrip(rstrip)
+
+                    return secret
+            except IOError:
+                _LOGGER.error(
+                    f'unable to read env value for {name} as a file.')
+                return None
+        else:
+            # unknown self.secrets_source value
```

Comment:
This comment is redundant right? The next line of code says pretty much the same thing